src/pkg/proxy/secret/authorizer.go

```diff
@@ -0,0 +1,63 @@
+//  Copyright Project Harbor Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package secret
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/goharbor/harbor/src/lib"
+)
+
+const (
+	secretPrefix = "Proxy-Cache-Secret"
+)
+
+// NewAuthorizer returns an instance of the authorizer
+func NewAuthorizer() lib.Authorizer {
+	return &authorizer{}
+}
+
+type authorizer struct{}
+
+func (s *authorizer) Modify(req *http.Request) error {
+	if req == nil {
+		return errors.New("the request is null")
+	}
+	repository, _, ok := lib.MatchManifestURLPattern(req.URL.Path)
```

Comment:
The nested if condition can be refactored to 
```
	if _, _, ok := MatchFunc1(); ok {

	}else if _, _, ok := MatchFunc2(); ok {

	}else if _, _, ok := MatchFunc3(); ok {

	}else {
		//not handler
		return
	}
	//handle it
```

Reply:
This authorizer is used when trying to push the artifact, this is in the client side, no artifact info in the context